### PR TITLE
More test coverage and other cleanup

### LINKED ISSUE #319: Check and document simulations for source spectra
Functionality really lives in pyradiosky but we should check that our simulations using various kinds of source spectra make sense.  A single memo documenting all the types (particularly spectral index and sub-band, which are the newest) would be nice to have before we declare this functionality fully operational.

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

I thought I'd added coverage for all the easy to cover lines in #449 but after it was merged I found that there were still some lines missing in simsetup.py. This PR does several things (sorry!):
- Adds coverage for some missing lines
- Updates the badges on the readme and other things throughout the repo to point to the main branch rather than master.
- Adds some text about pyradiosky in the tutorial and a link to the pyradiosky docs.
- Changes the RTD sphinx theme to match the one we use on other RASG repos.
- Runs one CI job on pip rather than conda. This has exposed some weirdnesses in the past on pyradiosky, so I wanted to do it here. I used the min_deps run because it's hard to get mpi4py to install via pip.
- Explicitly test against both Open MPI and MPICH implementations to ensure they both work and to avoid picking up the Intel MPI implementation (`impi`), which was breaking our tests.

## Motivation and Context

closes #319

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Reference simulation update or replacement
- [x] Documentation change (documentation changes only)
- [ ] Version change
- [x] Build or continuous integration change

## Checklist:

For all pull requests:
- [x] I have read the [contribution guide](CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Build or continuous integration change checklist:
- [x] If required or optional dependencies have changed (including version numbers), I have updated the readme to reflect this.
- [x] If this is a new CI setup, I have added the associated badge to the readme and to references/make_index.py (if appropriate).

Documentation change checklist:
- [x] Any updated docstrings use the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [x] If this is a significant change to the readme or other docs, I have checked that they are rendered properly on ReadTheDocs. (you may need help to get this branch to build on RTD, just ask!)

### CODE REVIEW

File: .github/workflows/testsuite.yaml [hunk 1]
```diff
@@ -63,54 +63,51 @@ jobs:
...
         run: |
-          pip install --no-deps .
+          python -m pytest -n auto --cov=pyuvsim --cov-config=.coveragerc --cov-report xml:./coverage.xml
 
-      - name: Run Tests
+      - name: check coverage report
         run: |
-          python -m pytest -n auto --cov=pyuvsim --cov-config=.coveragerc --cov-report xml:./coverage.xml --junitxml=test-reports/xunit.xml
+          ls
+          cat coverage.xml
```
Comment (mkolopanis) [inline .github/workflows/testsuite.yaml:104]: this feels like a bit of a debug item in the action list. Did we want to keep this?

**Review by mkolopanis — COMMENTED** [review]
Just one question!

File: .github/workflows/testsuite.yaml (hunk 1 above)
Comment (bhazelton) [reply .github/workflows/testsuite.yaml:104]: no, I think you're right it's a debugging thing

**Review by mkolopanis — APPROVED** [review]
yeah this works for me thanks!

**Pull request merged**